tooling/autodispose-error-prone-checker/src/test/java/com/uber/autodispose/error/prone/checker/MissingAutoDisposeErrorCheckerTest.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (C) 2017. Uber Technologies
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.uber.autodispose.error.prone.checker;
+
+import com.google.errorprone.CompilationTestHelper;
+import java.util.Arrays;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+@SuppressWarnings("CheckTestExtendsBaseClass")
+public class MissingAutoDisposeErrorCheckerTest {
+
+  @Rule public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+  private CompilationTestHelper compilationHelper;
+
+  @Before
+  public void setup() {
+    compilationHelper =
+        CompilationTestHelper
+            .newInstance(MissingAutoDisposeErrorChecker.class, getClass());
+    compilationHelper.setArgs(
+        Arrays.asList(
+            "-d",
+            temporaryFolder.getRoot().getAbsolutePath(),
+            "-XepOpt:AutoDisposeLeakCheck"
+                + "=com.uber.autodispose.error.prone.checker.ComponentWithLifeCycle"));
+  }
+
+  @Test
+  public void test_autodisposePositiveCases() {
+    compilationHelper.addSourceFile("MissingAutoDisposeErrorPositiveCases.java").doTest();
+  }
+
+  @Test
+  public void test_autodisposePositiveCases2() {
```

Comment:
Can we call this something more descriptive? Or at least have comments explaining what these two cases are verifying?

Reply:
https://github.com/uber/AutoDispose/pull/156/commits/47c35ce452fe3965ac92bf3240679f65dc5ffddc. Calling them default for default lifecycle and custom when input is provided.

Reply:
Cool, but please make this a separate commit next time. That commit's message/desc mentioned nothing about the descriptive method changes